rainbow-me/browser-extension
# feat: token search on chain (Cmd+K, swap and token details)

Fixes BX-1504, BX-1369
Figma link (if any):

## What changed (plus any additional context for devs)

Backend token search API doesn't return all tokens. If we don't detect any tokens from the backend API, we'll just search them on chain. This is now supported in swap page, Cmd+K menu and token details page.

## Screen recordings / screenshots

https://github.com/rainbow-me/browser-extension/assets/53529533/05078f32-dad5-4a2d-a3c4-6751117c4a04

## What to test

- Search for a random non supported backend token in swap page, and make sure the swap flow works.
- Open Cmd+K menu and search for a non supported backend token. You should be able to see it, and also have the ability to go to the token details page.

_You can easily see which token is not supported from backend by searching any contract address in swap page (production) and do the same thing in this branch and compare._

_For example a token that's not supported by the backend API is $PALAI (0x3Cb48AeB3D1abaDc23d2D8a6894B3A68338381C2)._
